### ISSUE TITLE
Android developerPayload not working in finishTransaction

### ISSUE THREAD

**Issue by CyberTron88** [dooboolab-community/react-native-iap#2751]
await finishTransaction({ purchase, isConsumable: false, developerPayloadAndroid: 'test123' });

When I did the above, and query the receipt based on the purchaseToken, the developerPayload is always empty.

**Comment by ng-ha** [comment]
I got the same issue. Did you fix that? @CyberTron88

**Comment by CyberTron88** [comment]
no. I actually wanted to use it to store our user id, end up storing using a different field.

**Comment by ng-ha** [comment]
me too. I ended up storing purchaseToken, because its kinda unique in subscription. @CyberTron88

**Comment by hyochan** [comment]
Could someone try `react-native-iap@next` module? Android billing client has been upgraded deprecating the older versions.

Also please indicate the `react-native-iap` version next time. Thank you! @CyberTron88

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.
